test/FormControlSpec.tsx

```diff
@@ -0,0 +1,157 @@
+      <FormControl type="color" data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('type').should.be.true;
+    element.getAttribute('type').should.equal('color');
+  });
+
+  it('should use controlId for id', () => {
+    const { getByTestId } = render(
+      <FormGroup controlId="foo">
+        <FormControl type="text" data-testid="test-id" />
+      </FormGroup>,
+    );
+
+    getByTestId('test-id').id.should.equal('foo');
+  });
+
+  it('should prefer explicit id', () => {
+    shouldWarn('ignored');
+
+    const { getByTestId } = render(
+      <FormGroup controlId="foo">
+        <FormControl type="text" id="bar" data-testid="test-id" />
+      </FormGroup>,
+    );
+
+    getByTestId('test-id').id.should.equal('bar');
+  });
+
+  it('should support ref forwarding', () => {
+    let input;
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormGroup controlId="foo">
+            <FormControl
+              type="text"
+              ref={(ref) => {
+                input = ref;
+              }}
+            />
+          </FormGroup>
+        );
+      }
+    }
+
+    render(<Container />);
+    input.tagName.toUpperCase().should.to.equal('INPUT');
+  });
+
+  it('should properly display size of FormControl', () => {
+    const { getByTestId } = render(
+      <FormControl type="text" size="lg" data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(2);
+    element.classList.contains('form-control-lg').should.be.true;
+  });
+
+  it('should properly display html size of FormControl', () => {
+    const { getByTestId } = render(
+      <FormControl type="text" htmlSize={42} data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('size').should.be.true;
+    element.getAttribute('size').should.equal('42');
+  });
+
+  it('Should have input as default component', () => {
+    const { getByTestId } = render(<FormControl data-testid="test-id" />);
+
+    getByTestId('test-id').tagName.toLowerCase().should.equal('input');
+  });
+
+  it('should support numbers as values', () => {
+    const { getByTestId } = render(
+      <FormControl
+        value={10}
+        onChange={() => undefined}
+        data-testid="test-id"
+      />,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('value').should.be.true;
+    element.getAttribute('value').should.equal('10');
+  });
+
+  it('should support an array of strings as values', () => {
+    const { getByTestId } = render(
+      <FormControl
+        value={['hello', 'world']}
+        onChange={() => undefined}
+        data-testid="test-id"
+      />,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('value').should.be.true;
+    element.getAttribute('value').should.equal('hello,world');
```

Comment:
Don't need hasAttribute since the value is checked on next line anyway

Reply:
Fixed in the commit https://github.com/react-bootstrap/react-bootstrap/pull/6224/commits/023f9b3e2cffbb507001f992d76bb9cfb4243310